.travis.yml

```diff
@@ -115,7 +122,8 @@ script:
     # also including their coverage may introduce to much noice. Concentrate on the coverage of local packages.
     # Execute go test on every local subpackage (resolved as dependencies) and generate covreage report for each.
     # Test packages pararell (xargs -P)
-    - go list ./... | grep -v /vendor/ | xargs -n1 -I {} -P 4 go test -v -covermode=atomic -coverprofile=../../../{}/coverage.txt {}
+
+    - if [ "$JOB_TYPE" = compile_and_basic_tests ]; then go list ./... | grep -v /vendor/ | xargs -n1 -I {} -P 4 go test -v -covermode=atomic -coverprofile=../../../{}/coverage.txt {}; fi
```

Comment:
I think you can actually write:

```
  - |
    if [ "$JOB_TYPE" = compile_and_basic_tests ]; then \
        go list ./... | grep -v /vendor/ | \
            xargs -n1 -I {} -P 4 \
                go test -v -covermode=atomic -coverprofile=../../../{}/coverage.txt {}; \
    fi
```

and maybe reformat other changed lines this way as well